contracts/FlowIDTableStaking.cdc

```diff
@@ -1822,6 +1822,91 @@ access(all) contract FlowIDTableStaking {
         return true
     }
 
+    /// Validates that a networking address is properly formatted
+    /// Requirements:
+    /// 1. Must not be an IP address
+    /// 2. Must contain a port number after a colon
+    /// 3. Must be a valid domain name format
+    access(all) view fun isValidNetworkingAddress(address: String): Bool {
+        // Check length
+       if  address.length == 0 && address.length > 510 {
+            return false
+        }
+
+        // Split the address into domain and port
+        let parts = address.split(separator: ":")
+        if parts.length != 2 {
+            return false
+        }
+
+        let domain = parts[0]
+        let port = parts[1]
+
+        // Check if port is a valid number between 1 and 65535
+        let portNum = UInt16.fromString(port)
+        if portNum == nil || portNum! < 1 || portNum! > 65535 {
+            return false
+        }
+
+        // Check if domain has at least one dot and valid characters
+        if !domain.contains(".") {
+            return false
+        }
+
+        // Check if domain contains only letters, digits and hyphens
+        let validChars = "ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789.-".utf8
+        for char in domain.utf8 {
+            if !validChars.contains(char) {
+                return false
+            }
+        }
```

Comment:
This might be quite inefficient: The lookup array of valid chars is re-built on each call of the function, and the loop ends up effectively performing a nested linear scan: first over all characters of the domain, and then again nested over all valid characters.

Reply:
@turbolent  Actually I suggested this one for `readability` mostly, didn't seem to be too inefficient at first sight. ( ~ maybe half efficient ) and I assumed this would be called once ofc.

Reply:
Ah, I see now! Nice 👍 Yeah, this is definitely more readable, the original implementation is not very obvious. I don't know how often this will be called. In case it is a heavily called function, we might want to trade some readability / "straight-forwardness" code for performance.

Reply:
This validation will only be called when a new node is registered or when the networking address of an existing node is update. Both of those actions don't happen frequently - once or twice every couple of epoch at most. Hence, I would rather have more readability than performance. Open to suggestions tho.